src/components/shapes/attributes.js

```diff
@@ -224,6 +225,50 @@ module.exports = templatedArray('shape', {
             '`config.editable` or `config.edits.shapePosition`.'
         ].join(' ')
     },
-
+    label: {
+        text: {
+            valType: 'string',
+            dflt: null,
+            editType: 'arraydraw',
+            description: 'Sets the text to display with shape.'
+        },
+        font: fontAttrs({
+            editType: 'calc+arraydraw',
+            colorEditType: 'arraydraw',
+            description: 'Sets the shape label text font.'
+        }),
+        position: {
+            valType: 'enumerated',
+            values: [
+                'top left', 'top center', 'top right',
+                'middle left', 'middle center', 'middle right',
+                'bottom left', 'bottom center', 'bottom right',
+                'top start', 'top end',
+                'middle start', 'middle end',
+                'bottom start', 'bottom end',
+            ],
+            editType: 'arraydraw',
+            description: 'Sets the position of the label relative to she shape.'
+        },
+        textangle: {
+            valType: 'angle',
+            dflt: 0,
+            editType: 'calc+arraydraw',
+            description: [
+                'Sets the angle at which the `text` is drawn',
+                'with respect to the horizontal.'
+            ].join(' ')
+        },
+        xanchor: annAttrs.xanchor,
+        yanchor: annAttrs.yanchor,
+        padding: {
+            valType: 'number',
+            dflt: 0,
```

Comment:
Perhaps we could/should increase this padding to 3px to be similar to the default padding used in other places e.g. bars.
```suggestion
            dflt: 3,
```

Reply:
Considered this -- I didn't do this initially because then if the user sets `padding` to 1 or 2 then the padding actually decreases rather than increases, which might be unexpected, but if other components use a default padding value it makes sense to do that here too.